internal/substatoptimizer/substatoptimizer.go

```diff
@@ -389,6 +389,12 @@ func RunSubstatOptim(simopt simulator.Options, verbose bool, additionalOptions s
 
 			substatGradients[idxSubstat] = substatEvalResult.DPS.Mean - initialMean
 
+			// fixes cases in which fav holders don't get enough crit rate to reliably proc fav (an important example would be fav kazuha)
+			// might give them "too much" cr (= max out liquid cr subs) but that's probably not a big deal
+			if charWithFavonius[idxChar] && idxSubstat == 1 {
```

Comment:
@k0l11 I think we want `if charWithFavonius[idxChar] && substat == core.CR {` to future proof us a bit better

Reply:
Sounds good, I've adjusted that part.